networkx/generators/lattice.py

```diff
@@ -145,25 +145,38 @@ def grid_graph(dim, periodic=False):
 
 @nx._dispatchable(graphs=None, returns_graph=True)
 def hypercube_graph(n):
-    """Returns the *n*-dimensional hypercube graph.
-
-    The nodes are the integers between 0 and ``2 ** n - 1``, inclusive.
+    """
+    Returns the n-dimensional hypercube graph.
 
-    For more information on the hypercube graph, see the Wikipedia
-    article `Hypercube graph`_.
+    The n-dimensional hypercube graph has 2^n nodes. Each node represents
+    a binary string of length 'n'. Two nodes are connected if their binary
+    representations differ in exactly one bit.
 
-    .. _Hypercube graph: https://en.wikipedia.org/wiki/Hypercube_graph
 
     Parameters
     ----------
     n : int
-        The dimension of the hypercube.
-        The number of nodes in the graph will be ``2 ** n``.
+        The dimension of the hypercube. Must be a non-negative integer.
+    create_using : Networkx graph constructor, optional (default: None)
+        Graph type to create. For example, 'nx.DiGraph' or 'nx.Graph'. If None,
+        a simple undirected Graph is used.
```

Comment:
This function does not take in `create_using`, the function signature is `def hypercube_graph(n):